lib/v2/screens/sign_up/signup_screen.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/v2/datasource/remote/api/signup_repository.dart';
+import 'package:seeds/v2/screens/sign_up/claim_invite/claim_invite_screen.dart';
+import 'package:seeds/v2/screens/sign_up/claim_invite/usecases/claim_invite_usecase.dart';
+import 'package:seeds/v2/screens/sign_up/display_name/display_name.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+
+class SignupScreen extends StatelessWidget {
+  const SignupScreen();
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (BuildContext context) => SignupBloc(
+        claimInviteUseCase: ClaimInviteUseCase(
+          signupRepository: SignupRepository(),
+        ),
+      ),
+      child: BlocConsumer<SignupBloc, SignupState>(
+        listener: (context, state) {},
+        builder: (context, state) {
```

Comment:
Here since you only listen to a `state.pageContent` for the switch-case you can put the `buildWhen` that only repaints when that variable changes.
Exmaple:
`buildWhen: (previus, current) =>previus.pageContent!=current.pageContent`
In order to avoid unnecessary rebuilds due to other states changes.